frontend-react/docs/data-fetching-patterns.md

```diff
@@ -49,55 +49,50 @@ Most of the work of actually making a particular request will live in hooks that
 
 React-query is a very full featured library that concerns itself with maintaining state for asynchronous actions, and providing good support around a cache for the results of those actions. While not specific to requests, react-query is built primarily to handle the complexities of making network requests within react, and that is what we use it for. For the most part, our patterns rely on two hooks provided by react-query, dealt with in more depth below. There is a lot more to react-query than this, including the QueryClient that allows for easy access to and manipulation of a request cache, and optimistic loading. The resources below should be a starting point for learning more about those.
 
-#### UseQuery
+#### UseQuery & rsUseQuery
 
-[useQuery](https://tanstack.com/query/v4/docs/reference/useQuery?from=reactQueryV3&original=https://react-query-v3.tanstack.com/reference/useQuery), in our app, is the hook used to fetch data.
+[useQuery](https://tanstack.com/query/v4/docs/reference/useQuery?from=reactQueryV3&original=https://react-query-v3.tanstack.com/reference/useQuery), in our app, is the hook used to fetch data. `rsUseQuery` is our very special, slightly wrapped version of `useQuery` that functions more or less exactly the same way as `useQuery`, while also allowing us to avoid making unathenticated requests before our Okta library has fully initialized.
 
-We pass useQuery a key ([see official docs for more](https://tanstack.com/query/v4/docs/guides/query-keys)), a function which will actually do the fetch, and an options object. On each render of the custom hook, useQuery will either run, and fetch data, or not, in the case that there is already fresh data available. React-query under the hood will periodically re-run the fetch function to get fresh data. We are running with default settings for this, but this behavior can be specified as well.
+`rsUseQuery` can be used the same way that tanstack's documentation indicates, with the one caveat being that while `useQuery` is very open to multiple ways to structure the arguments passed in, `rsUseQuery` hardens the options around call signatures. `rsUseQuery` will always be called with the signature:
```

Comment:
Did this name create any issues with React or the linter? It sounds like its use is as a hook, and they usually scream at you for not prefixing the hook name as `use`

Reply:
If it does scream at us, `useRSQuery` could work

Reply:
i haven't seen any linter screaming, though in retrospect i would have expected it. My guess is that whatever logic they have to determine whether something is a hook might be confused by all of the wrapping that's going on